### PR TITLE
Save as system #137

### LINKED ISSUE #137: A user/operator should be able to save a System as
Same as #105 but for systems

### PR DESCRIPTION
**Pull request by joelvdavies**
## Description

- Adds a menu to each system/subsystem with edit, save as and delete (which has been removed from the landing page)
- Adds save as dialog

image
image

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

Closes #137

### CODE REVIEW

File: cypress/e2e/system.cy.ts [hunk 1]
```diff
@@ -272,49 +285,148 @@ describe('System', () => {
...
+            location: 'System location',
+            owner: 'System owner',
+            importance: 'medium',
+            parent_id: null,
+          })
+        );
+      });
+    });
+
+    it("save as a system eidting only a system's name (in subsystem)", () => {
```
Comment (MatteoGuarnaccia5) [inline cypress/e2e/system.cy.ts:353]:
```suggestion
    it("save as a system editing only a system's name (in subsystem)", () => {
```

**Review by MatteoGuarnaccia5 — COMMENTED** [review]
image

When moving a system into another system, I found that the chosen system still remains visible, until you refresh the screen or click into the system it was moved into. Should probably refetch the systems after moved so that it displays the correct systems

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]
Everything work well. Just one minor comment:

- can you make it so that when a user "save as" a system that the name defaults with `_copy_{iteration of the copy}`. Just so it matches the implementation done in catalogue items table view.

**Comment by joelvdavies** [comment]
> Everything work well. Just one minor comment:
> 
> * can you make it so that when a user "save as" a system that the name defaults with `_copy_{iteration of the copy}`. Just so it matches the implementation done in catalogue items table view.

Just realised this is actually easier to resolve with the changes to use the table, I will just create conflicts here so I will make a note in the issue and do it there if that's alright (although I have just refactored slightly here to reduce code repetition)

**Review by joshuadkitenge — APPROVED** [review]

**Pull request merged**